syntax = "proto3";

package streaming;

message StreamInfo {
    string session_id = 1;
    string media = 2;
}
  
message StreamState {
    enum State {
      // The default state.
      UNDEFINED = 0;
      // The call is newly created.
      NEW = 1;
      // The call is connected.
      ACTIVE = 6;
      // The call is finished.
      ENDED = 7;
    }
    State state = 2;
}
  
message StreamRequest {
    string url = 1;
}
  
message StreamResponse {
    oneof stream_response {
        StreamInfo stream_info = 1;
        StreamState stream_state = 2;
    } 
}

service Streaming {
    rpc StreamingRTSP(StreamRequest) returns (stream StreamResponse) {}
}
